Array: Inverse of an array

Op-panda opened the pull request:
This is inverse of an array problem ,quiet famous and important one. please do merge this problem

<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes 

### Description
Inverse of an array problem, 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3043

Op-panda committed:
Added inverse of an array - FIXED #3043

This is a famous problem. It is written with proper explaination and test cases. please merge this ISSUE#3043

HarshCasper commented on C-Plus-Plus/ds/inverse of an array.cpp:
```diff
@@ -0,0 +1,50 @@
+
+    Sample Output
+    0 1 2
+    */
```
Not properly indented. Check the existing code to understand how to do that

HarshCasper commented on C-Plus-Plus/ds/inverse of an array.cpp:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    return 0;
+    }
```
Format the Code properly and indent it

HarshCasper commented on C-Plus-Plus/ds/inverse of an array.cpp:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    for(int i =0 ; i<n; i++){
+        cout<<inv[i]<<" ";      //output the inverse array
```
Never use comments beside the code. Use it above the line of code

HarshCasper commented on C-Plus-Plus/ds/inverse of an array.cpp:
```diff
@@ -0,0 +1,50 @@
+    /*If the array elements are swapped with their corresponding indices,
+      the array finally results is inverse of the array.
+    */
+
```
Add a more intuitive description

Op-panda committed:
Delete inverse of an array.cpp

Op-panda committed:
ARRAY: Inverse of an array

A meaningful array question , that can be useful to understand the array concept and enhance logic building
I have done changes that was requested. Hope it is now good.

Op-panda committed:
Update README.md

Op-panda committed:
Update README.md

Op-panda committed:
Update README.md

Op-panda committed:
Update README.md

Commit:
Merge remote-tracking branch 'upstream/master' into Op-panda-3043

Op-panda committed:
Update README.md

Op-panda commented:
@HarshCasper Now please Review this I have fixed the changes you requested. please do a review once more and tell if I need to do some more changes??

Op-panda committed:
Update inverse of an array.cpp

Op-panda committed:
Update README.md

Op-panda committed:
Update inverse of an array.cpp

HarshCasper commented on C-Plus-Plus/README.md:
```diff
@@ -122,6 +122,7 @@
 - [Ascending Priority Queue] (ds/Ascending_priority_queue.cpp)
 - [Priority_Queue_using_Linked_List](ds/Priority_Queue_using_Linked_List.cpp)
 - [Maximum Width of Binary Tree](ds/Maximum_Width_BinaryTree.cpp)
+- [Inverse of an array](C-Plus-Plus/ds/inverse of an array.cpp)
```
This will not work. Fix it!

HarshCasper replied:
This will not work. Fix it!

HarshCasper commented on C-Plus-Plus/ds/inverse of an array.cpp:
```diff
@@ -0,0 +1,73 @@
+/*
```
Name your file as `inverse_of_array.cpp`

Op-panda replied:
do i need to make a different PR with another file name. please help out. I dont know how to do that

HarshCasper reviewed (changes requested):
Indent and format the code as well properly

HarshCasper reviewed (changes requested):
Indent and format the code as well properly

Op-panda committed:
Update README.md

Op-panda committed:
Rename inverse of an array.cpp to inverse_of_an_array.cpp

Op-panda committed:
Update README.md

Op-panda committed:
Update README.md

Op-panda committed:
Update README.md

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,73 @@
+so assume a blank array of same size. Now at 4th index, element in inverse array will be 0. 
+Similarly, at 0th index, element in inverse array will be 1
+and similar swapping for the rest of elemnts 
+At last you will be getting 1,4,2,3,0 in inverse array. 
```
Remove this

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,73 @@
+3)Third line contains elements of the original array.
+
+Output:
+1)contains t outputs, inverse array of each original array. */
```
Remove this

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,73 @@
+
+#include<bits/stdc++.h>
+using namespace std;
+int main(){
```
Keep the main code logic outside the `main()` function. Use `main()` only for the driver code

HarshCasper replied:
Not yet resolved @Op-panda

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,73 @@
+    cout<<inv[i]<<" ";      
+    }
+}
+return 0;
```
None of the code here is either formatted or indented. How come you resolved my comments?

Op-panda replied:
I am sorry. I didn't know about those. This is my first contibution.

Op-panda committed:
Update inverse_of_an_array.cpp

Op-panda committed:
Update inverse_of_an_array.cpp

Op-panda committed:
Update inverse_of_an_array.cpp

Op-panda committed:
Update inverse_of_an_array.cpp

Op-panda committed:
Update inverse_of_an_array.cpp

Op-panda commented:
Is now the changes ok ??

Op-panda committed:
Update inverse_of_an_array.cpp

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,65 @@
+/*
+INVERSE OF AN ARRAY   
+/*
```
Remove the `/*`

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,65 @@
+// give input for ori array.
+// create a temporary variable then, and store the element from the original array.
+// In inv array for each value of temp, store the index of ori array.
+// Display the inverse array.
```
Remove this

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,65 @@
+    }
+    return 0;
+}
+// Driver code ends
```
Remove this

HarshCasper commented on C-Plus-Plus/ds/inverse_of_an_array.cpp:
```diff
@@ -0,0 +1,65 @@
+1 4 2 3 0
+0 1 2
+*/
+
```
Remove this blank line

Op-panda committed:
Update inverse_of_an_array.cpp

HarshCasper committed:
Merge branch 'master' into Op-panda-3043
